sensor/NetworkMonitorSensor.js

```diff
@@ -0,0 +1,426 @@
+      const scheduledKey = `${key}-${monitorType}`;
+      if ( scheduledKey in this.sampleJobs ) {
+        log.warn(`${monitorType} on ${key} already started`);
+      } else {
+        log.debug(`scheduling sample job ${monitorType} on ${key} ...`);
+        this.sampleJobs[scheduledKey] = this.scheduleSampleJob(monitorType,ip,cfg[monitorType]);
+        log.debug(`scheduling process job ${monitorType} on ${key} ...`);
+        this.processJobs[scheduledKey] = this.scheduleProcessJob(monitorType,ip,cfg[monitorType]);
+      }
+    }
+  }
+
+  stopMonitorDevice(key) {
+    log.info(`stop monitoring ${key} ...`)
+    for ( const monitorType of MONITOR_TYPES ) {
+      const scheduledKey = `${key}-${monitorType}`;
+      if ( scheduledKey in this.sampleJobs ) {
+        log.debug(`UNscheduling ${monitorType} on ${key} ...`);
+        clearInterval(this.sampleJobs[scheduledKey]);
+        delete(this.sampleJobs[scheduledKey])
+      } else {
+        log.warn(`${monitorType} on ${key} NOT scheduled`);
+      }
+      if ( scheduledKey in this.processJobs ) {
+        log.debug(`UNscheduling ${monitorType} on ${key} ...`);
+        clearInterval(this.processJobs[scheduledKey]);
+        delete(this.processJobs[scheduledKey])
+      } else {
+        log.warn(`${monitorType} on ${key} NOT scheduled`);
+      }
+    }
+  }
+
+  stopMonitorDeviceAll() {
+    Object.keys(this.sampleJobs).forEach( scheduledKey => {
+      log.debug(`UNscheduling ${scheudledKey} in sample jobs ...`);
+      clearInterval(this.sampleJobs[scheduledKey]);
+      delete(this.sampleJobs[scheduledKey]);
+    })
+    Object.keys(this.processJobs).forEach( scheduledKey => {
+      log.debug(`UNscheduling ${scheudledKey} in stat jobs ...`);
+      clearInterval(this.processJobs[scheduledKey]);
+      delete(this.processJobs[scheduledKey]);
+    })
+    log.info("after stop all: ", JSON.stringify(this.sampleJobs,null,4));
+  }
+
+  async applyPolicyDevice(host, state, cfg) {
+    try {
+      log.info(`Apply policy on device ${host} with state(${state}) and config(${cfg}) ...`);
+      const key = host.o.mac;
+      switch ( state ) {
+        case true: {
+          // always restart to run with latest config
+          this.stopMonitorDevice(key);
+          this.startMonitorDevice(key, host.o.ipv4Addr, cfg);
+          break;
+        }
+        case false: {
+          this.stopMonitorDevice(key);
+          break;
+        }
+        default: {
+          log.error("unsupported state: ",state);
+        }
+      }
+    } catch (err) {
+      log.error(`failed to apply policy on device ${host}: `,err);
+    }
+    return;
+  }
+
+  async applyPolicy(host, ip, policy) {
+    log.info(`Apply network monitor policy with host(${host}), ip(${ip}), policy(${policy})`);
+    try {
+        if (ip === '0.0.0.0') {
+            if (policy.state === true) {
+                if (fc.isFeatureOn(FEATURE_NETWORK_MONITOR, false)) {//compatibility: new firewlla, old app
```

Comment:
no need this backward compatible code any more, it was only used for super old features

Reply:
Removed as suggested

Reply:
removed as suggested